scheyaml_test.go

```diff
@@ -61,3 +78,49 @@ func TestSchemaToNode_ReturnsExpectedOutput(t *testing.T) {
 	// If the properties are as expected, test the comments
 	assert.Equal(t, string(expectedData), string(actualData))
 }
+
+func TestSchemaToNode_InvalidSchema(t *testing.T) {
+	t.Parallel()
+	// Arrange
+	inputData, _ := os.ReadFile(path.Join("testdata", "test-schema-nested-pattern-properties.json"))
+
+	compiler := jsonschema.NewCompiler()
+	schema, err := compiler.Compile(inputData)
+	require.NoError(t, err)
+
+	var overrides map[string]any
+	overridesData, err := os.ReadFile(path.Join("testdata", "test-schema-nested-pattern-properties-overrides.yaml"))
+	require.NoError(t, err)
+	require.NoError(t, yaml.Unmarshal(overridesData, &overrides))
+
+	// Act
+	result, err := SchemaToNode(schema, WithOverrideValues(overrides))
+
+	// Assert
+	expected := &InvalidSchemaError{}
+	require.ErrorAs(t, err, &expected)
```

Comment:
Can any interesting assertions be made on the actual result error? If not, then this is fine